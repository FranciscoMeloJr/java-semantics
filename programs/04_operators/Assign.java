class main {

	void printAll(String[] vs, int[] v) {
		for (int i = 0; i<v.length; i++) {
      System.out.println(vs[i]+"  =>  "+v[i]);
		}
	}

  int nrOps = 12;

  String[] initVS() {
    String[] vs = new String[nrOps];
    int i=0;

		vs[i++] = "4 = 2   ";
		vs[i++] = "4 +=   2";
		vs[i++] = "4 -=   2";
		vs[i++] = "4 *=   2";
		vs[i++] = "4 /=   2";
		vs[i++] = "4 &=   2";
		vs[i++] = "4 |=   2";
		vs[i++] = "4 ^=   2";
		vs[i++] = "4 %=   2";
		vs[i++] = "4 <<=  2";
		vs[i++] = "4 >>=  2";
		vs[i++] = "4 >>>= 2";

    return vs;
  }

	static void main(String[] args) {
    int[] v = new int[nrOps];
    String[] vs = initVS();

    for(int i=0; i<nrOps; i++) {
      v[i] = 4;
    }
    int i=0;

		v[i++] = 2;
		v[i++] += 2;
		v[i++] -= 2;
		v[i++] *= 2;
		v[i++] /= 2;
		v[i++] &= 2;
		v[i++] |= 2;
		v[i++] ^= 2;
		v[i++] %= 2;
		v[i++] <<= 2;
		v[i++] >>= 2;
		v[i++] >>>= 2;

    printAll(vs,v);
    System.out.println("Done!");
	}
}
