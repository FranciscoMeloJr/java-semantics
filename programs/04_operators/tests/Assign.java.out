4 = 2     =>  2
4 +=   2  =>  6
4 -=   2  =>  2
4 *=   2  =>  8
4 /=   2  =>  2
4 &=   2  =>  0
4 |=   2  =>  6
4 ^=   2  =>  6
4 %=   2  =>  0
4 <<=  2  =>  16
4 >>=  2  =>  1
4 >>>= 2  =>  1
Done!
